File: LMS.Microservices/LMS.Microservices/lms-frontend/src/types/enrollment.ts
export interface CreateEnrollmentDto {
  studentId: number
  courseId: number
}

export interface StudentEnrollment {
  id: number
  courseId: number
  courseName: string
  enrollmentDate: Date
  status: EnrollmentStatus
}

export enum EnrollmentStatus {
  Pending = 'Pending',
  Active = 'Active',
  Completed = 'Completed',
  Cancelled = 'Cancelled',
}

export interface Enrollment {
  id: number
  studentId: number
  courseId: number
  studentName: string
  courseName: string
  enrollmentDate: Date
  status: EnrollmentStatus
  prerequisites?: PrerequisiteStatus[]
}
export interface EnrollmentDto {
  id: number
  studentId: number
  courseId: number
  enrollmentDate: Date
  status: EnrollmentStatus
}

export interface CreateEnrollmentDto {
  studentId: number
  courseId: number
}

export interface PrerequisiteStatus {
  prerequisiteId: number
  courseName: string
  isMet: boolean
  isMandatory: boolean
}

export interface EnrollmentValidationResult {
  isValid: boolean
  message?: string
  missingPrerequisites?: PrerequisiteStatus[]
}

File: LMS.Monolith/LMS.Core/Entities/Enrollment.cs
namespace LMS.Core.Entities
{
    // LMS.Core/Entities/Enrollment.cs
    public class Enrollment
    {
        public int Id { get; set; }
        public int StudentId { get; set; }
        public int CourseId { get; set; }
        public DateTime EnrollmentDate { get; set; }
        public EnrollmentStatus Status { get; set; }
        public Student Student { get; set; }
        public Course Course { get; set; }
    }
}
